### PR TITLE
 Add ability to specify custom logging module 

### PR DESCRIPTION
**Pull request by ntenczar**
In the user's application's config.exs, specify the logging module that
should be used by GenRetry.

For example:

```
config :gen_retry, GenRetry.Logger, logger: GenRetry.Utils
```

This will use the default GenRetry.Utils module. The module specified
must conform to the GenRetry.Logger behavior.

### CODE REVIEW

File: lib/gen_retry/utils.ex [hunk 1]
```diff
@@ -0,0 +1,9 @@
+defmodule GenRetry.Utils do
+  require Logger
+
+  @behaviour GenRetry.Logger
+
+  def log(message) do
```
Comment (andyleclair) [inline lib/gen_retry/utils.ex:7]: please add `@impl true`

File: lib/test_logger.ex [hunk 2]
```diff
@@ -0,0 +1,7 @@
+defmodule GenRetry.TestLogger do
+  @behaviour GenRetry.Logger
+
+  def log(_message) do
```
Comment (andyleclair) [inline lib/test_logger.ex:5]: `@impl true` here also

**Review by andyleclair — COMMENTED** [review]
2 comments, and ya gotta fix the `mix.exs` conflict

**Comment by mogorman** [comment]
:+1:

**Review by gamache — APPROVED** [review]

**Pull request merged**